Mlem/Views/Shared/Instance/InstanceView+Logic.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  InstanceView+Logic.swift
+//  Mlem
+//
+//  Created by Sjmarf on 03/02/2024.
+//
+
+import SwiftUI
+
+extension InstanceView {
+    
+    // swiftlint:disable:next function_body_length
+    func attemptToLoadInstanceData() {
+        if instance.administrators == nil {
+            Task {
+                do {
+                    if let url = URL(string: "https://\(instance.name)") {
+                        let info = try await apiClient.loadSiteInformation(instanceURL: url)
+                        DispatchQueue.main.async {
+                            withAnimation(.easeOut(duration: 0.2)) {
+                                instance.update(with: info)
+                            }
+                        }
+                    } else {
+                        errorDetails = ErrorDetails(title: "\"\(instance.name)\" is an invalid URL.")
+                    }
+                } catch let APIClientError.decoding(data, error) {
+                    withAnimation(.easeOut(duration: 0.2)) {
+                        if let content = String(data: data, encoding: .utf8) {
+                            if content.contains("<div class=\"kbin-container\">") {
+                                errorDetails = ErrorDetails(
+                                    title: "KBin Instance",
+                                    body: "We can't display KBin instance details.",
```

Comment:
All of our other text is passive voice

```suggestion
                                    body: "KBin instances are not currently supported.",
```